Sources/SmithyReadWrite/ReadingClosure.swift

```diff
@@ -83,101 +83,82 @@ public func timestampReadingClosure<Reader: SmithyReader>(format: TimestampForma
     }
 }
 
-public extension String {
```

Comment:
Made the following changes on ReadingClosures & WritingClosures:
- Removed them as extensions on the types they read/write; instead they are static methods.  This prevents them from polluting the namespace of Swift and Foundation types.